LFS-330: Installing multiple vocabularies in OBO format causes subsequent installations to copy data from previous installations

OrdiNeu opened the pull request:
[Jira link](https://phenotips.atlassian.net/browse/LFS-330)

To test: from the sidebar, head to the Vocabularies admin page and install HP, followed by TAXRANK. Querying the entries of TAXRANK via `/Vocabularies/TAXRANK.1.json` will no longer show all of the terms from HP.

OrdiNeu committed:
LFS-330: Installing multiple vocabularies in OBO format causes
subsequent installations to copy data from previous installations

sdumitriu commented on modules/vocabularies/src/main/java/ca/sickkids/ccm/lfs/vocabularies/internal/OboParser.java:
```diff
@@ -51,6 +51,7 @@
 @Component(
     service = SourceParser.class,
```
```suggestion
    service = SourceParser.class,
    scope=ServiceScope.PROTOTYPE,
```

I wonder if just adding this line makes it work, without the need to change anything else.

OrdiNeu replied:
This also seems to work, though I'm more in favour of `InheritableThreadLocal` since it's also the approach currently being used by `ca.sickkids.ccm.lfs.vocabularies.internal.OwlParser` (i.e. we should either adopt it in both cases and forego `InheritableThreadLocal` or use `InheritableThreadLocal` in both cases).

OrdiNeu committed:
LFS-330: Installing multiple vocabularies in OBO format causes
subsequent installations to copy data from previous installations
Switch to using prototype scope classes.

sdumitriu commented on modules/vocabularies/src/main/java/ca/sickkids/ccm/lfs/vocabularies/internal/OboParser.java:
```diff
@@ -50,7 +51,9 @@
     service = SourceParser.class,
+    scope = ServiceScope.PROTOTYPE,
     name = "SourceParser.OBO")
+@SuppressWarnings("checkstyle:ClassDataAbstractionCoupling")
```
Is this still needed?
```suggestion
```

sdumitriu committed:
[misc] Removed no longer needed checkstyle suppression

sdumitriu commented:
Ah, my bad, my proposed smaller fix doesn't actually fix anything since the same instance is used.

sdumitriu commented:
I manually reverted the other commit, your original fix was the right one.

OrdiNeu commented:
> Ah, my bad, my proposed smaller fix doesn't actually fix anything since the same instance is used.
I tested the smaller fix -- it was working on my machine. Was there a test case where it failed?
